ext/crypto/import_key.rs

```diff
@@ -685,12 +793,81 @@ fn import_key_ec(
           }
         }
       };
-      ImportKeyResult::Ec {
+      Ok(ImportKeyResult::Ec {
         raw_data: RawKeyData::Public(data),
-      }
+      })
     }
-    _ => return Err(unsupported_format()),
-  })
+    KeyData::Pkcs8(data) => {
+      match named_curve {
+        EcNamedCurve::P256 => {
+          let secret_key = p256::SecretKey::from_pkcs8_der(&data).unwrap();
+
+          let point =
+            secret_key.public_key().as_affine().to_encoded_point(false);
+
+          // 3.
+          if point.is_identity() {
+            return Err(data_error("Invalid key data"));
+          }
+        }
+        EcNamedCurve::P384 => {
+          //@todo(sean-wykes) Validate P384 secret-key on import(pkcs8)
+          /*let secret_key =
+            p384::SecretKey::from_pkcs8_der(&raw_data.data).unwrap();
+
+          let point =
+            secret_key.public_key().as_affine().to_encoded_point(false);
+          // 3.
+          if point.is_identity() {
+            return Err(type_error("Invalid key data".to_string()));
+          }*/
+
+          // for now, just use PKCS8 bytes
+        }
+      };
+
+      Ok(ImportKeyResult::Ec {
+        raw_data: RawKeyData::Public(data),
+      })
+    }
+    KeyData::Spki(data) => {
+      let pk_info: spki::SubjectPublicKeyInfo =
+        spki::SubjectPublicKeyInfo::try_from(data.as_ref()).unwrap();
+      let pk = pk_info.subject_public_key;
```

Comment:
Same goes for SPKI. There should be a check here that checks the algorithm OID and the namedCurve are correct. Look at how this is done for RSA-PSS (checking OID and parameters).

Reply:
OK. Thanks for the hint. I've been focussing on functionality first .. now getting into the validation stuff. Just committed for SPKI, PKCS8 next on list.